lib/controllers/carplay_controller.dart

```diff
@@ -50,16 +50,9 @@ class FlutterCarplayController {
       if (value) {
         for (var template in templateHistory) {
           switch (template) {
-            // case final CPTabBarTemplate tabBarTemplate:
-            //   for (final (tabIndex, tab) in tabBarTemplate.templates.indexed) {
-            //     if (tab.uniqueId == elementId) {
-            //       tabBarTemplate.templates[tabIndex] = updatedTemplate;
-            //       return;
-            //     }
-            //   }
             case final CPInformationTemplate informationTemplate:
               if (informationTemplate.uniqueId == elementId) {
-                template = updatedTemplate;
+                template = updatedTemplate as CPTemplate;
```

Comment:
`updatedTemplate` is of type `CPInfomationTemplate` which derives from `CPTemplate`, so a type conversation should not be needed

Reply:
Right, removed it.